### PR TITLE
vectorscan: Submission; rspamd: Use vectorscan, add hyperscan variant

### PR DESCRIPTION
**Pull request by essandess**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: mail/rspamd/Portfile [hunk 1]
```diff
@@ -74,11 +74,23 @@ depends_lib-append  port:fann \
...
                     port:redis
 
-# Match requirements of dependency hyperscan.
-supported_archs     x86_64
+variant hyperscan \
+    description {Use hyperscan instead of vectorscan.} {
+    depends_lib-replace \
+                    port:vectorscan \
+                    port:hyperscan
+    supported_archs x86_64
```
Comment (ryandesign) [inline mail/rspamd/Portfile:82]: Does `supported_archs` work in a variant?

**Review by ryandesign — COMMENTED** [review]
Looks like vectorscan was previously submitted in #23468.

File: mail/rspamd/Portfile (hunk 1 above)
Comment (essandess) [reply mail/rspamd/Portfile:82]: I'm not sure. What I do see is that `port install rspamd +hyperscan` on `arm64` compiles the entire dependency chain from scratch, which doesn't look like desired behavior to me, even if the ultimate binary can be linked to an `x86_64` binary and work [?]

I suggest using `known_fail` for this variant on `arm64`.

**Comment by essandess** [comment]
This PR is ready to merge.

File: mail/rspamd/Portfile (hunk 1 above)
Comment (ryandesign) [reply mail/rspamd/Portfile]: If you mean that all of the dependencies get rebuilt with the universal variant, then that is working correctly. To build something (rspamd) for x86_64 on arm64, dependencies need to be built universal. I would recommend not adding code to the Portfile to subvert how MacPorts is intended to work.

Comment (essandess) [reply mail/rspamd/Portfile]: Please note that your original comment has already been addressed: `supported_arch`s is not used in a variant. 

Also, `hyperscan` only supports `x86_64`, which is the entire motivation of the request for this PR in https://trac.macports.org/ticket/69717.

To the best of my knowledge, this Portfile is configured correctly for the upstream ports and is ready to merge.

**Pull request merged**